src/fare-contracts/components/ConsumeCarnetBottomSheet.tsx

```diff
@@ -0,0 +1,87 @@
+import {getAxiosErrorMetadata} from '@atb/api/utils';
+import {Confirm} from '@atb/assets/svg/mono-icons/actions';
+import {
+  BottomSheetContainer,
+  useBottomSheet,
+} from '@atb/components/bottom-sheet';
+import {Button} from '@atb/components/button';
+import {MessageInfoBox} from '@atb/components/message-info-box';
+import {GenericSectionItem, Section} from '@atb/components/sections';
+import {ThemeText} from '@atb/components/text';
+import {StyleSheet} from '@atb/theme';
+import {consumeCarnet} from '@atb/ticketing';
+import {FareContractTexts, useTranslation} from '@atb/translations';
+import React, {useState} from 'react';
+import {ScrollView} from 'react-native-gesture-handler';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+
+type Props = {
+  fareContractId: string;
+};
+
+export const ConsumeCarnetBottomSheet = ({fareContractId}: Props) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<boolean>(false);
+  const {close} = useBottomSheet();
+
+  const onConsume = async () => {
```

Comment:
IIRC, we want to start using react-query for new API operations, so we should do that also for this one.

Reply:
I think this is the one type of requests that shouldn't use react-query. There is no state to keep up to date here, and no dependencies to keep track of. Just an user interaction that causes an API request, which doesn't need to be more complicated that this.

Reply:
Fair enough! 👍

Reply:
We should use react-query. Then we get error/loading statues, retry possibility, and more out of the box, and we get the same typing/behaviour on all network requests.